src/main/java/net/snowflake/ingest/streaming/internal/ParquetTypeGenerator.java

```diff
@@ -94,73 +53,94 @@ static ParquetTypeInfo generateColumnParquetTypeInfo(ColumnMetadata column, int
     Type.Repetition repetition =
         column.getNullable() ? Type.Repetition.OPTIONAL : Type.Repetition.REQUIRED;
 
-    // Handle differently depends on the column logical and physical types
-    switch (logicalType) {
-      case FIXED:
-        parquetType = getFixedColumnParquetType(column, id, physicalType, repetition);
-        break;
-      case ARRAY:
-      case OBJECT:
-      case VARIANT:
-        // mark the column metadata as being an object json for the server side scanner
-        metadata.put(id + ":obj_enc", "1");
-        // parquetType is same as the next one
-      case ANY:
-      case CHAR:
-      case TEXT:
-      case BINARY:
-        parquetType =
-            Types.primitive(PrimitiveType.PrimitiveTypeName.BINARY, repetition)
-                .as(LogicalTypeAnnotation.stringType())
-                .id(id)
-                .named(name);
-        break;
-      case TIMESTAMP_LTZ:
-      case TIMESTAMP_NTZ:
-      case TIMESTAMP_TZ:
-        parquetType =
-            getTimeColumnParquetType(
-                column.getScale(),
-                physicalType,
-                logicalType,
-                TIMESTAMP_SUPPORTED_PHYSICAL_TYPES,
-                repetition,
-                id,
-                name);
-        break;
-      case DATE:
-        parquetType =
-            Types.primitive(PrimitiveType.PrimitiveTypeName.INT32, repetition)
-                .as(LogicalTypeAnnotation.dateType())
-                .id(id)
-                .named(name);
-        break;
-      case TIME:
-        parquetType =
-            getTimeColumnParquetType(
-                column.getScale(),
-                physicalType,
-                logicalType,
-                TIME_SUPPORTED_PHYSICAL_TYPES,
-                repetition,
-                id,
-                name);
-        break;
-      case BOOLEAN:
-        parquetType =
-            Types.primitive(PrimitiveType.PrimitiveTypeName.BOOLEAN, repetition).id(id).named(name);
-        break;
-      case REAL:
-        parquetType =
-            Types.primitive(PrimitiveType.PrimitiveTypeName.DOUBLE, repetition).id(id).named(name);
-        break;
-      default:
+    if (column.getSourceIcebergDataType() != null) {
```

Comment:
for next PR: add validation that we only ever see column.sourceIcebergDataType as non-null when isIcebergMode is true. I saw you're already checking if this field is null on any column when we're in iceberg mode, need similar check for non-iceberg-mode.